src/twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +109,271 @@ def test_copyAliasedSender(self):
+        o = parseOptions("-odb")
+        self.assertTrue(o.background)
+
+
+    def test_foreGroundDelivery(self):
+        """
+        The I{-odf} flags specifies foreground delivery.
+        """
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions("-odf")
+        self.assertFalse(o.background)
+
+
+    def test_recipientsFromHeaders(self):
+        """
+        The I{-t} flags specifies that recipients should be obtained
+        from headers.
+        """
+        stdin = NativeStringIO(
+            'To: Curly <invaliduser2@example.com>\n'
+            'Cc: Larry <invaliduser1@example.com>\n'
+            'Bcc: Moe <invaliduser3@example.com>\n'
+            '\n'
+            'Oh, a wise guy?\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions("-t")
+        self.assertEqual(len(o.to), 3)
+
+
+    def test_setFrom(self):
+        """
+        When a message has no I{From:} header, a I{From:} value can be
+        specified with the I{-F} flag.
+        """
+        stdin = NativeStringIO(
+            'To: invaliduser2@example.com\n'
+            'Subject: A wise guy?\n\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions(["-F", "Larry <invaliduser1@example.com>", "-t"])
+        self.assertEqual(o.sender, "Larry <invaliduser1@example.com>")
+
+
+    def test_overrideFromFlagByFromHeader(self):
+        """
+        The I{-F} flag specifies the From: value.  However, I{-F} flag is
+        overriden by the value of From: in the e-mail header.
+        """
+        stdin = NativeStringIO(
+            'To: Curly <invaliduser4@example.com>\n'
+            'From: Shemp <invaliduser4@example.com>\n')
+        self.patch(sys, 'stdin', stdin)
+        o = parseOptions(["-F", "Groucho <invaliduser5@example.com>", "-t"])
+        self.assertEqual(o.sender, "invaliduser4@example.com")
+
+
+    def test_run(self):
+        """
+        Call L{mailmail.run}, and specify I{-oep} to print errors
+        to stderr.  The sender, to, and printErrors options should be
+        set and there should be no failure.
+        """
+        argv = ("test_mailmail.py", "invaliduser2@example.com", "-oep")
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'argv', argv)
+        self.patch(sys, 'stdin', stdin)
+        mailmail.run()
+        self.assertEqual(self.options.sender, mailmail.getlogin())
+        self.assertEqual(self.options.to, ["invaliduser2@example.com"])
+        # We should have printErrors set because we specified "-oep"
+        self.assertTrue(self.options.printErrors)
+        # We should not have any failures.
+        self.assertIsNone(mailmail.failed)
+
+    if platformType == "win32":
+        test_run.skip = ("mailmail.run() does not work on win32 due to "
+                         "lack of support for getuid()")
+
+
+    def test_readInvalidConfig(self):
+        """
+        Read an invalid configuration file.  Error messages
```

Comment:
Minor comment
`Read an invalid configuration file.`  for me does not add any new information on top of why I can understand by reading `def test_readInvalidConfig(self):`

Maybe:

```
Error messages for illegal UID value, illegal GID value, and illegal identity entry are sent to stderr.
```